### DISCUSSION TITLE
Fast reverb for data augmentation

### DISCUSSION THREAD

**Discussion by desh2608** [lhotse-speech/lhotse#787] [Ideas]
Currently, Lhotse supports reverberating a cut/recording by explicitly providing a room impulse response (see `CutSet.reverb_rir()`). We can have an alternate functionality that generates a room impulse response on-the-fly, using the following method: https://github.com/yluo42/FRA-RIR.

Basically, this is an approximation of the image source method which is widely used to simulate RIRs. Because of the approximations, it is fast even when computing on CPU. This would be useful for on-the-fly augmentation in the data-loaders.

**Comment by pzelasko** [comment]
Great idea!

**Comment by desh2608** [comment]
Implemented in https://github.com/lhotse-speech/lhotse/pull/799. Note that at the moment, the method does not support simulation of multi-channel arrays.
